executor/adapter.go

```diff
@@ -734,15 +734,6 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx sessionctx.Context, p plannerco
 		return false, nil
 	}
 
-	// check txn
-	txn, err := ctx.Txn(false)
```

Comment:
Make we check also check that the session is not in a transaction by `SessionVars.InTxn`

Reply:
good catch